### ISSUE TITLE
Nullability tracking is broken for `Nullable<T>` with awaits

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#76886]
**Version Used**: 
Version 17.14.0 Preview 1.0 [35717.212.main]

**Steps to Reproduce**:
```C#
MyStruct? myStruct = await GetStructAsync();
MyStruct s1 = myStruct.Value;

static async Task<MyStruct?> GetStructAsync()
{
    await Task.Yield();
    return new MyStruct();
}

struct MyStruct { }
```

**Diagnostic Id**:

n/a

**Expected Behavior**:
This should produce `warning CS8629: Nullable value type may be null.` when accessing `.Value`.

**Actual Behavior**:
No warning.

**Comment by Metalaka** [comment]
I investigated this issue and found that the `NullableWalker` was incorrectly marking the result of an `await` expression as `NotNull` when the result type is a **NullableValueType** (e.g., `Task<int?>` awaiting to `int?`).

The root cause was in `NullableWalker.VisitAwaitExpression`, which unconditionally called `SetNotNullResult(node)` for any result that is a `ValueType`. This failed to account for `Nullable<T>`, which can indeed be null after an `await`.

### Technical Analysis
The issue is located in `NullableWalker.VisitAwaitExpression`. The current implementation has the following check:

```csharp
if (node.Type.IsValueType || node.HasErrors || awaitableInfo.GetResult is null)
{
    SetNotNullResult(node);
}
```

Since `node.Type.IsValueType` is `true` for `Nullable<T>`, the compiler unconditionally marks the result of the `await` as `NotNull` whenever it is a value type. This is incorrect for `Nullable<int>`, `Nullable<bool>`, etc., which are still "nullable" and can be `null` after being awaited from a `Task<T?>`.

### Reproduction
The following test case demonstrates the bug:
```csharp
[Fact]
public async Task AwaitedNullableValueTypeAssignedToNonNullable_ProducesCS8629()
{
    var src = @"
#nullable enable
using System.Threading.Tasks;
public class TestClass
{
    public async Task Main()
    {
        Task<int?> GetNullableIntAsync() => Task.FromResult<int?>(null);
        
        int? result = await GetNullableIntAsync();
        int nonNullableInt = result.Value; // Should warn CS8629: Nullable value type may be null.
    }
}";

    var comp = CreateCompilation(src);
    comp.VerifyDiagnostics(
        // (11,30): warning CS8629: Nullable value type may be null.
        //          int nonNullableInt = result.Value; // warn expected
        Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "result")
            .WithLocation(11, 30));
}
```
